sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/query.sql

```diff
@@ -14,48 +14,134 @@ WITH new_profile_ping AS (
+      SAFE_OFFSET(0)
+    ] AS attribution_medium,
+    ARRAY_AGG(environment.settings.attribution.source RESPECT NULLS ORDER BY submission_timestamp)[
+      SAFE_OFFSET(0)
+    ] AS attribution_source,
+    ARRAY_AGG(environment.settings.attribution.ua RESPECT NULLS ORDER BY submission_timestamp)[
+      SAFE_OFFSET(0)
+    ] AS attribution_ua,
+    ARRAY_AGG(
+      environment.settings.default_search_engine_data.load_path RESPECT NULLS
+      ORDER BY
+        submission_timestamp
+    )[SAFE_OFFSET(0)] AS engine_data_load_path,
+    ARRAY_AGG(
+      environment.settings.default_search_engine_data.name RESPECT NULLS
+      ORDER BY
+        submission_timestamp
+    )[SAFE_OFFSET(0)] AS engine_data_name,
+    ARRAY_AGG(
+      environment.settings.default_search_engine_data.origin RESPECT NULLS
+      ORDER BY
+        submission_timestamp
+    )[SAFE_OFFSET(0)] AS engine_data_origin,
+    ARRAY_AGG(
+      environment.settings.default_search_engine_data.submission_url RESPECT NULLS
+      ORDER BY
+        submission_timestamp
+    )[SAFE_OFFSET(0)] AS engine_data_submission_url,
+    ARRAY_AGG(environment.system.apple_model_id RESPECT NULLS ORDER BY submission_timestamp)[
+      SAFE_OFFSET(0)
+    ] AS apple_model_id,
+    ARRAY_AGG(metadata.geo.city RESPECT NULLS ORDER BY submission_timestamp)[
+      SAFE_OFFSET(0)
+    ] AS city,
+    ARRAY_AGG(metadata.geo.db_version RESPECT NULLS ORDER BY submission_timestamp)[
+      SAFE_OFFSET(0)
+    ] AS db_version,
+    ARRAY_AGG(metadata.geo.subdivision1 RESPECT NULLS ORDER BY submission_timestamp)[
+      SAFE_OFFSET(0)
+    ] AS subdivision1,
+    ARRAY_AGG(normalized_channel RESPECT NULLS ORDER BY submission_timestamp)[
+      SAFE_OFFSET(0)
+    ] AS normalized_channel,
+    ARRAY_AGG(normalized_country_code RESPECT NULLS ORDER BY submission_timestamp)[
+      SAFE_OFFSET(0)
+    ] AS country,
+    ARRAY_AGG(normalized_os RESPECT NULLS ORDER BY submission_timestamp)[
+      SAFE_OFFSET(0)
+    ] AS normalized_os,
+    ARRAY_AGG(normalized_os_version RESPECT NULLS ORDER BY submission_timestamp)[
+      SAFE_OFFSET(0)
+    ] AS normalized_os_version,
+    ARRAY_AGG(
+      payload.processes.parent.scalars.startup_profile_selection_reason RESPECT NULLS
+      ORDER BY
+        submission_timestamp
+    )[SAFE_OFFSET(0)] AS startup_profile_selection_reason,
+    ARRAY_AGG(environment.settings.attribution.dltoken RESPECT NULLS ORDER BY submission_timestamp)[
+      SAFE_OFFSET(0)
+    ] AS attribution_dltoken,
+    ARRAY_AGG(
+      environment.settings.attribution.dlsource RESPECT NULLS
+      ORDER BY
+        submission_timestamp
+    )[SAFE_OFFSET(0)] AS attribution_dlsource,
   FROM
     `moz-fx-data-shared-prod.telemetry.new_profile`
   WHERE
     {% if is_init() %}
       DATE(submission_timestamp) >= '2010-01-01'
       {% if parallel_run() %}
-        AND sample_id = {sample_id}
+        AND sample_id = {sample_id }
```

Comment:
This is weird.  Should this be `{{sample_id}}` or `@sample_id` instead?  Though I don't see how any of `{sample_id}`, `{{sample_id}}`, or `@sample_id` would work as currently written, and I also don't see any `parallel_run()` macro available in this context.

@lucia-vargas-a it looks you added this recently in #3962.

Reply:
Hey, the formatting is correct! This code runs as part of improvements to the initialization command currently in review in [PR-4322](https://github.com/mozilla/bigquery-etl/pull/4322).

Reply:
If we keep this Python string formatting parameter syntax then `bqetl format` would need to be changed to not break it (it's currently automatically adding the extra space.

However, IMO it would be better to avoid this syntax:
* `{sample_id}` (or `{mapped_values}` as #4322 currently has it) would be better as a query parameter like `@sample_id` or `@mapped_values`.  This is safer, and avoids having to manually SQL-quote string values going in.
* Instead of ``INSERT INTO `{project_id}.{dataset_id}.{table_id}` `` you could add `--append-table=true` and `--replace=false` parameters to the `bq query` command when parallelizing the table init, which will have the same effect without requiring that kludge.

(I'll also add PR comments to that effect in #4322)